### PR TITLE
[core] Fix placement groups scheduling when no resources are specified

### PR DESCRIPTION
**Pull request by vitsai**
Before, actors scheduled with no resources and placement groups would not be placed with the placement group or bundle. Fix this by using the phantom bundle resource that is also used for the Ready() call as an additional placement group constraint. (See https://github.com/ray-project/ray/pull/16819 for context)

## Related issue number

https://github.com/ray-project/ray/issues/34866

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by vitsai** [comment]
Bug fix: https://github.com/ray-project/ray/pull/40038

**Review by rkooo567 — COMMENTED** [review]
Can you post the ray status output when there's a pending task that requires a placement gruop (just to see the output does not regress due to this change we always add bundles resources 0.0001)

File: src/ray/common/bundle_spec.cc
Comment (vitsai) [reply src/ray/common/bundle_spec.cc:198]: good point! removed

File: python/ray/tests/test_placement_group.py
Comment (vitsai) [reply python/ray/tests/test_placement_group.py]: Yep!

**Comment by vitsai** [comment]
Here is output with both pending and running pg tasks:

```
======== Autoscaler status: 2023-10-05 07:13:09.740863 ========
GCS request time: 0.000660s

Node status
---------------------------------------------------------------
Active:
 1 node_7dedc912c22c07df0fd49dc5b6e8450472310ad4041ddd2540e519dc
Idle:
 (no idle nodes)
Pending:
 (no pending nodes)
Recent failures:
 (no failures)

Resources
---------------------------------------------------------------
Total Usage:
 4.0/36.0 CPU (4.0 used of 8.0 reserved in placement groups)
 0B/35.23GiB memory
 0B/17.61GiB object_store_memory

Total Demands:
 {'CPU': 1.0}: 6+ pending tasks/actors (6+ using placement groups)

Node: 7dedc912c22c07df0fd49dc5b6e8450472310ad4041ddd2540e519dc
 Usage:
  4.0/36.0 CPU (4.0 used of 8.0 reserved in placement groups)
  0B/35.23GiB memory
  0B/17.61GiB object_store_memory
 Activity:
  Resource: CPU currently in use.
  Busy workers on node.
  Resource: CPU_group_0_38c7049e01961594ddbd48ae247806000000 currently in use.
  Resource: CPU_group_38c7049e01961594ddbd48ae247806000000 currently in use.
```

I think the only potential change needed is in node activity and that can be a follow up as it is not super noisy.

**Review by rkooo567 — APPROVED** [review]
LGTM. I think we should merge this after we merge https://github.com/ray-project/ray/pull/40038.

**Comment by rkooo567** [comment]
ping me after ^

**Comment by vitsai** [comment]
This looks ready to merge; same failures as the stacked PR

File: src/ray/core_worker/common.h
Comment (jjyao) [inline src/ray/core_worker/common.h:111]: This change is already merged?

**Pull request merged**
